pkg/checkmarx/checkmarx.go

```diff
@@ -246,37 +246,16 @@ func sendRequestInternal(sys *SystemInstance, method, url string, body io.Reader
 		defer closer.Close()
 	}
 	response, err := sys.client.SendRequest(method, fmt.Sprintf("%v/cxrestapi%v", sys.serverURL, url), requestBody, header, nil)
-	if err != nil {
+	if err != nil && !piperutils.ContainsInt(acceptedErrorCodes, response.StatusCode) {
```

Comment:
For a StatusCode of 200...399, err would be nil?

Reply:
yes, see https://github.com/SAP/jenkins-library/blob/master/pkg/http/http.go#L332

The original implementation overlooked that a `404` always returns an error and the handling of the accepted codes came too late ...